server/handlers/clients.js

```diff
@@ -190,14 +184,19 @@ function addConfigHandler(req, res) {
   const id = req.params.id;
 
   if (clients.get(id))
-    return res
-      .status(409)
-      .send({ message: `A client with the id '${id}' already exists` });
+    return res.status(409).send({
+      error: {
+        message: `A client with the id '${id}' already exists`,
+        code: 409
+      }
+    });
 
+  req.type = 'add';
```

Comment:
`type` is a very general thing to decorate the request with, maybe namespace it in an object so that it cannot collide with other plugins or middleware?

Reply:
Good call with the name spacing. I was worried about the name too but just plugged it in to get the support.